go/libraries/doltcore/schema/alterschema/dropcolumn.go

```diff
@@ -108,7 +110,16 @@ func DropColumn(ctx context.Context, tbl *doltdb.Table, colName string, foreignK
 	if err != nil {
 		return nil, err
 	}
-	newTable, err := doltdb.NewTable(ctx, vrw, schemaVal, rd, indexData)
+
+	var autoVal types.Value
```

Comment:
Seems like maybe this logic should be a method on the Table type

`func (t *table) UpdateSchema(...) *table` or similar?